test/stream.js

```diff
@@ -238,20 +236,30 @@ describe('gulp-ng-config', function () {
             expect(file.contents.toString()).to.equal(expectedOutputB.toString());
           }));
 
-      // tests output with no matching environment key
-      streamC.pipe(plugin('gulp-ng-config', {
-        environment: 'nonExistant'
-      }))
-          .pipe(through.obj(function (file) {
-            expect(file.contents.toString()).to.equal(expectedOutputC.toString());
-          }));
-
-      es.merge(streamA, streamB, streamC)
+      es.merge(streamA, streamB)
           .pipe(through.obj(function () {
             done();
           }));
     });
 
+    it('should emit an error if an environment key is supplied and the key does not exist', function (done) {
+      var stream = gulp.src(path.normalize(__dirname + '/mocks/input_3.json')),
+          spy = chai.spy();
+      expect(function () {
+        stream.pipe(plugin('gulp-ng-config', {
+          environment: 'env'
+        })).on('error', function () {
+          spy();
+          this.emit('end');
+        }).on('end', function () {
+          expect(spy).to.have.been.called();
+          done();
+        }).pipe(through.obj(function () {
+          done();
```

Comment:
Shouldn't this test verify one or the other here, rather than having both `done` statements?